### PR TITLE
Move documentation testing to a plugin

### PR DESCRIPTION
**Pull request by nik9000**
This makes it easier for other projects to use it. This also cleans up some issues I found while getting it working on "other projects".

### CODE REVIEW

**Comment by nik9000** [comment]
@rjernst can you have a look at this one?

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/doc/DocsTestPlugin.groovy [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * under the License.
+ */
+package org.elasticsearch.gradle.doc
+
+import org.elasticsearch.gradle.test.RestTestPlugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+
+/**
+ * Sets up tests for documetation.
```
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/doc/DocsTestPlugin.groovy]: documetation -> documentation

File: docs/build.gradle [hunk 2]
```diff
@@ -17,68 +17,38 @@
...
-    List<String> languages = [
-      'js', 'json', // These languages should almost always be marked autosense
-      'sh', 'shell', // These are often curl commands that should be converted
-    ]
-    if (false == languages.contains(it.language)) {
-      return
-    }
-    println(it)
-  }
+project.buildRestTests.docs = fileTree(project.projectDir) {
```
Comment (rjernst) [inline docs/build.gradle]: should be no need for the `project.` prefix here

**Comment by rjernst** [comment]
LGTM

**Comment by shavo007** [comment]
What is the plugin!?

**Comment by nik9000** [comment]
> What is the plugin!?

Sorry, it is a gradle plugin. We access it like:
```
apply plugin: 'elasticsearch.docs-test'
```

If you feel the need to use Elasticsearch's build as a dependency for your own gradle builds then this is it's maven location: http://search.maven.org/#artifactdetails%7Corg.elasticsearch.gradle%7Cbuild-tools%7C5.0.0-alpha2%7Cjar

You might want to do that if you are developing an Elasticsearch plugin. We have relatively fancy tools for that in the gradle build.

I don't think this particular change is released to that though.

**Comment by shavo007** [comment]
Cool!

**Pull request merged**